### PR TITLE
Renames proxies

### LINKED ISSUE #953: Rename proxies
Rename:

- `VariableView` -> `ElementArrayView` (including file name!)
- `*Proxy` -> `*View` (and files, e.g. coords_proxy_test)
- `DataProxy` -> `DataArrayView` (same for `DataConstProxy`)

### PR DESCRIPTION
**Pull request by dtasev**
Renames:
VariableView -> ElementArrayView (and filename)
DataProxy -> DataArrayView
DataConstProxy -> DataArrayConstView
*Proxy -> *View

Closes #953

### CODE REVIEW

File: common/include/scipp/common/span/test/catch.hpp [hunk 1]
```diff
@@ -4757,7 +4757,7 @@ namespace Catch {
 
 // start catch_fatal_condition.h
 
-// start catch_windows_h_proxy.h
+// start catch_windows_h_view.h
```
Comment (SimonHeybrock) [inline common/include/scipp/common/span/test/catch.hpp]: `span` is not our code but a subproject (https://github.com/tcbrindle/span), can you undo the changes in that folder?

Comment (dtasev) [reply common/include/scipp/common/span/test/catch.hpp]: Done. (apparently you can revert files from GitHub Files Changed tab now)
